.ci/updatecli.d/bump-upstream-agent-version.yml

```diff
@@ -0,0 +1,66 @@
+name: Upgrade the upstream agent version and related transitive dependencies
+# This task
+# - checks if the current otel-agent version referenced by the distro is the latest release
+# - if not, it will update the agent version AND the semconv and contrib versions to match the upstream
+
+scms:
+  githubConfig:
+    kind: github
+    spec:
+      user: '{{ requiredEnv "GITHUB_ACTOR" }}'
+      owner: elastic
+      repository: elastic-otel-java
+      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
+      username: '{{ requiredEnv "GITHUB_ACTOR" }}'
+      branch: main
+      commitusingapi: true
+
+sources:
+
+  upstream-agent-version:
+    kind: githubrelease
+    spec:
+      owner: open-telemetry
+      repository: opentelemetry-java-instrumentation
+      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
+      typefilter:
+        latest: true
+    transformers:
+      - trimprefix: "v"
+
+  distro-otel-agent-version:
+    kind: toml
+    scmid: githubConfig
+    spec:
+      file: gradle/libs.versions.toml
+      key: versions.opentelemetryJavaagentAlpha
+
+conditions:
+  update-required:
+    disablesourceinput: true
+    kind: shell
+    spec:
+      command: test "{{ source "upstream-agent-version" }}-alpha" != "{{ source "distro-otel-agent-version" }}"
+
+targets:
+
+  update-agent-version:
+    name: "Update upstream OpenTelemetry agent version and related dependencies"
+    conditionids: [update-required]
```

Comment:
What's the reason for this condition? IIUC, conditions will be the ones deciding whether the targets can be processed or no.

Reply:
We only want to perform an update if the current version isn't up to date.
That's why I though we would need the condition, don't we?

Reply:
That's the default behaviour, you don't need to be explicit as far as I recall.

For instance:
- https://github.com/elastic/opbeans-java/blob/main/.ci/updatecli.d/bump-version.yml#L44-L59